internal/pkg/model/component.go

```diff
@@ -168,11 +169,11 @@ func (c *ComponentsMap) doSet(component *Component) {
 func (c *ComponentsMap) addDefaultBucketPrefix(component *Component) {
 	r := regexpcache.MustCompile(`(?i)[^a-zA-Z0-9-]`)
 	bucketPrefix := fmt.Sprintf(`%s.c-%s-`, component.Data.DefaultBucketStage, r.ReplaceAllString(component.Id, `-`))
-	c.defaultBucketPrefixes[bucketPrefix] = component.Id
+	c.defaultBucketPrefixes[component.Id] = bucketPrefix
```

Comment:
Přehodil jsem klíče a hodnoty. Když se prohledávají ty prefixy tak se musí stejně iterovat přes celou mapu než se najde ten správný a takhle se dá aspoň prefix bucketu podle id komponenty vzít rovnou.

Reply:
"spravne" by sa to malo riesit s 2-ma map. Potom sa v oboch pripadoch zaobides bez for + ziskas hodnotu za konstantny cas `O(1)`.

Je to tak spravene napr. v `Naming`:
https://github.com/keboola/keboola-as-code/blob/ef7e1e8ab2a7e01731bcadd4f6b33bb780f9b120/internal/pkg/model/naming.go#L49-L50

Nie ze by na tom nejak extra zalezalo (rychlost, ...), ale bude to trocha prehladnejsie.

Reply:
Ale v `MatchDefaultBucketInTableId()` to nejde bez for, protože se musí porovnat jestli tableId začíná jedním z uložených prefixů a ne že se mu rovná (nebo jsem aspoň nenašel žádnou built-in metodu která by to zvládla).

Reply:
No mohol by si ziskat najprv potencionalny prefix, tak ho vyhladat v `map`, nie?
Ale to je okrajova vec, kludne to nechaj, tak ako to je.

Reply:
Tomu asi nerozumím. 🤔  Na vstupu do `MatchDefaultBucketInTableId()` přijde `tableId` (tzn. např. `in.c-keboola-ex-db-mysql-123.accounts`), tak jak zjistím potenciální prefix? Jo počkej, že bych z toho odstranil název tabulky (vše za druhou tečkou) a potom id (vše za poslední pomlčkou) a to co zbyde (v tomto případě `in.c-keboola-ex-db-mysql-`) rovnou zkusil jako klíč namísto zkoušení `strings.HasPrefix()` na jeden prefix za druhým. To by vlastně šlo.